src/ja/server/database/database.py

```diff
@@ -0,0 +1,99 @@
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import List, Callable
+from ja.common.job import Job
+from ja.server.database.types.job_entry import DatabaseJobEntry
+from ja.server.database.types.work_machine import WorkMachine
+
+
+class ServerDatabase(ABC):
```

Comment:
There can also be a method for querying all jobs on a specific working machine.

Reply:
I'll add this as a parameter to query_jobs.

Reply:
Also probably a method for getting all workmachines with enough available resources(to run a job), the RunningWorkMachineResources may be passed as an argument.

Reply:
> Also probably a method for getting all workmachines with enough available resources(to run a job), the RunningWorkMachineResources may be passed as an argument.

This is something the scheduler will do internally.